src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsHostBuilderExtensions.cs

```diff
@@ -34,6 +35,19 @@ public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, str
             return AddApplicationInsights(builder, instrumentationKey, (_, level) => level > LogLevel.Debug, null);
         }
 
+        /// <summary>
+        /// Registers Application Insights and <see cref="ApplicationInsightsLoggerProvider"/> with an <see cref="IHostBuilder"/>。
+        /// </summary>
+        /// <param name="builder">The host builder.</param>
+        /// <param name="instrumentationKey">The Application Insights instrumentation key.</param>
+        /// <param name="samplingSettings">Sampling settings</param>
+        /// <param name="snapshotCollectorConfiguration">Snapshot collector configuration.</param>
+        /// <returns>A <see cref="IHostBuilder"/> for chaining additional operations.</returns>
+        public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, string instrumentationKey, SamplingPercentageEstimatorSettings samplingSettings, SnapshotCollectorConfiguration snapshotCollectorConfiguration)
```

Comment:
I want to revisit how we pass in these configuration parameters (to allow binding to `IConfiguration`). It doesn't have to happen in this PR, but wanted to call it out.

Reply:
Let's discuss it, I'd love to revisit it and use IConfiguration and/or ApplicaitonInsights configuration options class.
I recall functions parse configuration and pass settings to the webjobs to configure AI, so whatever we change should work well with functions too.

Reply:
We're in the process of taking the DI changes we've worked on into the functions runtime right now. Once that is done, we'll be using a consistent configuration model everywhere and will be able to use either IConfiguration or the configuration options.